public/app/features/alerting/unified/components/rule-editor/FolderWithoutGroup.tsx

```diff
@@ -0,0 +1,199 @@
+            <Stack direction="row" alignItems="center">
+              {(!isCreatingFolder && (
+                <>
+                  <Controller
+                    render={({ field: { ref, ...field } }) => (
+                      <div style={{ width: 420 }}>
+                        <RuleFolderPicker
+                          inputId="folder"
+                          invalid={!!errors.folder?.message}
+                          {...field}
+                          enableReset={true}
+                          onChange={({ title, uid }) => {
+                            field.onChange({ title, uid });
+                            resetGroup();
+                          }}
+                        />
+                      </div>
+                    )}
+                    name="folder"
+                    rules={{
+                      required: { value: true, message: 'Select a folder' },
+                    }}
+                  />
+                  <Text color="secondary">
+                    <Trans i18nKey="alerting.rule-form.folder.new-folder-or">or</Trans>
+                  </Text>
+                  <Button
+                    onClick={onOpenFolderCreationModal}
+                    type="button"
+                    icon="plus"
+                    fill="outline"
+                    variant="secondary"
+                    disabled={!contextSrv.hasPermission(AccessControlAction.FoldersCreate)}
+                    data-testid={selectors.components.AlertRules.newFolderButton}
+                  >
+                    <Trans i18nKey="alerting.rule-form.folder.new-folder">New folder</Trans>
+                  </Button>
+                </>
+              )) || (
+                <div>
+                  <Trans i18nKey="alerting.rule-form.folder.creating-new-folder">Creating new folder</Trans>
+                  {'...'}
+                </div>
+              )}
+            </Stack>
+          </Field>
+        }
+        {isCreatingFolder && (
+          <FolderCreationModal onCreate={handleFolderCreation} onClose={() => setIsCreatingFolder(false)} />
+        )}
+      </Stack>
+
+      {isCreatingFolder && (
+        <FolderCreationModal onCreate={handleFolderCreation} onClose={() => setIsCreatingFolder(false)} />
+      )}
+    </div>
+  );
+}
+
+function FolderCreationModal({
+  onClose,
+  onCreate,
+}: {
+  onClose: () => void;
+  onCreate: (folder: Folder) => void;
+}): React.ReactElement {
+  const styles = useStyles2(getStyles);
+
+  const [title, setTitle] = useState('');
+  const onSubmit = async () => {
+    const newFolder = await createFolder({ title: title });
+    if (!newFolder.uid) {
+      appEvents.emit(AppEvents.alertError, ['Folder could not be created']);
+      return;
+    }
+
+    const folder: Folder = { title: newFolder.title, uid: newFolder.uid };
+    onCreate(folder);
+    appEvents.emit(AppEvents.alertSuccess, ['Folder Created', 'OK']);
+  };
+
+  const error = containsSlashes(title);
```

Comment:
I think slashes are no longer issue for GMA

Reply:
good point, this is from old code moved here, but let's remove it 👍

Reply:
I would remove this logic in another PR, wdyt?